collator/src/lib.rs

```diff
@@ -447,6 +450,54 @@ fn run_collator_node<S, P, Extrinsic>(
 	Ok(service)
 }
 
+/// Build a collator node with the given `RelayChainContext` and `ParachainContext`
+/// build by the given `BuildParachainContext` and arguments to the underlying polkadot node.
+///
+/// This function does not block. It returns an `AbstractService` that can be used as a future
+pub async fn build_collator<P, B>(
```

Comment:
@bkchr maybe it would be better named start_collator in the end since it awaits for it

Reply:
I've done